src/platforms/stm32/cmake/version_from_git.cmake

```diff
@@ -0,0 +1,58 @@
+#
+# This file is part of AtomVM.
+#
+# Copyright 2024 Winford (Uncle  Grumpy) <winford@object.stream>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+#
+
+function( version_from_git )
+
+  # Find Git
+  find_package(Git)
+  if(!Git_FOUND)
+    message( FATAL_ERROR "Git not found" )
```

Comment:
I did not consider that specific possibility, but there is no way exposed from libopencm3 to determine the version used for the build.  If a source code release of libopencm3 was used, then the version returned would be `{undefined,undefined,undefined}`.

Reply:
We must make sure build is still possible without a .git directory. e.g. sources from a tarball, but unknown or something similar must be returned.

Reply:
We already have `git` as a build requirement:

https://www.atomvm.net/doc/main/build-instructions.html#building-for-stm32

So this check should not fail. If the build is using a tarball download of libopencm3 the `os_version` will just be undefined, since there is no other way of determining which set of sources is being used. There is no release information or versioning in the libopencm3 source tree.